lib/ecam.c

```diff
@@ -0,0 +1,1066 @@
+/*
+ *      The PCI Library -- Direct Configuration access via PCIe ECAM
+ *
+ *      Copyright (c) 2023 Pali Rohár <pali@kernel.org>
+ *
+ *      Can be freely distributed and used under the terms of the GNU GPL.
+ */
+
+/*
+ * Tell 32-bit platforms that we are interested in 64-bit variant of off_t type
+ * as 32-bit variant of off_t type is signed and so it cannot represent all
+ * possible 32-bit offsets. It is required because off_t type is used by mmap().
+ */
+#define _FILE_OFFSET_BITS 64
```

Comment:
I'm thinking if it would not be better to move all physical address mappings into own file and de-duplicate ecam and mmio-ports code. _FILE_OFFSET_BITS then would be needed to set only at that own file as mmap is not used on any other place.

Reply:
Yes, please.

Reply:
Ok, I can look at it later as I see it needs more code refactor.